crates/miden-tx/src/executor/mod.rs

```diff
@@ -360,31 +360,41 @@ where
         .map_err(TransactionExecutorError::TransactionProgramExecutionFailed);
 
         match result {
-            Ok(_) => Ok(NoteAccountExecution::Success),
-            Err(tx_execution_error) => {
+            Ok(_) => Ok(NoteConsumptionInfo::new_successful(
+                tx_inputs
+                    .into_input_notes()
+                    .into_iter()
+                    .map(|note| note.into_note())
+                    .collect::<Vec<_>>(),
```

Comment:
We already have `input_notes` local variable, right? If so, we should be able to use it directly instead of using `tx_inputs.into_input_notes()`.

Reply:
`input_notes` is a shared reference so I would have to perform a `note.note().clone()` instead of `note.into_note()` here.